src/core/scene/a-scene.js

```diff
@@ -299,6 +334,56 @@ module.exports = registerElement('a-scene', {
     },
 
     /**
+     * Behavior-updater meant to be called from scene render.
+     * Abstracted to a different function to facilitate unit testing (`scene.tock()`) without
+     * needing to render.
+     */
+    tock: {
+      value: function (time, timeDelta) {
+        var systems = this.systems;
+
+        // Components.
+        this.behaviors.tock.forEach(function (component) {
+          if (!component.el.isPlaying) { return; }
+          component.tock(time, timeDelta);
+        });
+        // Systems.
+        Object.keys(systems).forEach(function (key) {
+          if (!systems[key].tock) { return; }
+          systems[key].tock(time, timeDelta);
+        });
+      }
+    },
+
+    /**
+     * Behavior-updater meant to be called from scene after it renders.
+     * Abstracted to a different function to facilitate unit testing (`scene.tickPostRender()`) without
+     * needing to render. timeDelta is the time it took to render the scene.
+     */
+    tickPostRender: {
+      value: function (time, timeDelta) {
+        // Components.
+        this.behaviors.tickPostRender.forEach(function (component) {
```

Comment:
hmmm, not all behaviors are handled the same. The post behaviors don't care if the component's el isPlaying, I think we said that it shouldn't matter for these? As it is right now even if the scene is not playing the base rendering still happens so post render should as well